api/types.proto

```diff
@@ -30,20 +30,20 @@ message Annotations {
 	repeated IndexEntry indices = 4 [(gogoproto.nullable) = false];
 }
 
-message GenericString {
+message NamedGenericResource {
 	string kind = 1;
 	string value = 2;
 }
 
-message GenericDiscrete {
+message DiscreteGenericResource {
 	string kind = 1;
 	int64 value = 2;
 }
 
 message GenericResource {
 	oneof resource {
```

Comment:
I don't think it would make sense to use `Any` here. The scheduler needs to understand these so it can perform accounting of resource availability on nodes.